### ISSUE TITLE
iPhone client fails to connect

### ISSUE THREAD

**Issue by mfat** [tailscale/tailscale#5938]
### What is the issue?

My iPhone fails to connect. My PC on the same wifi network connects to tailscale and works well. 
It shows needs authentication but tapping the button won't do anything.
When I connect to a proxy server tailscale successfully connects.

### Steps to reproduce

Open the iOS app and tap activate. 

### OS

iOS

### OS version

15.6.1

### Bug report

BUG-b15e0f6add65277a58a64e58fff17f5a6f217d3783f402b78e7a84b783efa97d-20221014105849Z-46a3c5dad1239a07

**Comment by bradfitz** [comment]
Does your iPhone work on LTE without your home LAN/ISP? What proxies/MITM are going on in your home network?

**Comment by mfat** [comment]
> Does your iPhone work on LTE without your home LAN/ISP? What proxies/MITM are going on in your home network?

It does. My wireless router is configured to use DNS-over-https. All desktop machines on this network connect to Tailscale without a problem. Only my iPhone and Android phones don't.

**Comment by bradfitz** [comment]
You have an http proxy (privoxy) and explicitly mention that your wireless router is doing DNS-over-HTTPS. Was there something that motivated you setting up both those things, like your ISP regularly blocking some of your traffic?

(I'm trying to figure out what class of problem this bug is about: is it a bug in our code, or is that something's messing with our traffic?)

**Comment by mfat** [comment]
Yes I live somewhere with very restricted internet censorship and dns is heavily manipulated.
